core/src/main/java/org/elasticsearch/node/internal/InternalSettingsPreparer.java

```diff
@@ -114,10 +114,9 @@
             }
             if (loadFromEnv) {
                 for (String allowedSuffix : ALLOWED_SUFFIXES) {
-                    try {
-                        settingsBuilder.loadFromPath(environment.configFile().resolve("elasticsearch" + allowedSuffix));
-                    } catch (SettingsException e) {
-                        // ignore
+                    Path path = environment.configFile().resolve("elasticsearch" + allowedSuffix);
+                    if (Files.exists(path)) {
```

Comment:
I don't like Files.exists in general but personally I think this is the right decision for a good surgical fix!